pkg/engine/lifecycletest/pulumi_test.go

```diff
@@ -3466,107 +3466,6 @@ func TestDeletedWith(t *testing.T) {
 	assert.Len(t, snap.Resources, 0)
 }
 
-func TestDeletedWithCircularDependency(t *testing.T) {
```

Comment:
This test fails now that `VerifyIntegrity` actually checks `deletedWith`. I'm minded to suggest that this means this test represents behaviour we do not want to see/shouldn't be possible to trigger and so have tentatively removed it, but definitely open to better suggestions if this is a legitimate use case (though if it is, the changes to `VerifyIntegrity` will have also broken those use cases).

Reply:
Is it that now we can never have a circular dependency in `DeletedWith`, because of the way we're deleting those? Or could this result in more VerifyIntegrity failures in the wild?

It's somewhat worrying that this test is now failing with the stricter `VerifyIntegrity` calls. Could there be snapshots in the wild that would end up failing the integrity check now? We probably shouldn't have errors for users with existing snapshots :thinking:

Reply:
I don't think programs like this could ever have actually been constructed before. You can't get a URN until it's ran its step and written to the snapshot.